src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -347,19 +345,13 @@ let command ~root ~backend =
     ( fswatch
     , [ "-r"
       ; root
-      ; (* If [inotify_special_path] is not passed here, then the [--exclude
-           _build] makes fswatch not descend into [_build], which means it never
-           even discovers that [inotify_special_path] exists. This is despite
-           the fact that [--include] appears before. *)
-        inotify_special_path
```

Comment:
What was this removed? The comment doesn't seem stale

Reply:
We don't actually have any tests for the fswatch path and I don't think we plan to add any. So it's just dead code.

Reply:
It does mean that if someone tries to use the passive watch mode on a machine where the fswatch backend is used, `dune` will just hang. That doesn't seem great. It's not harmful, so I think we should just leave it.

Reply:
Okay, I'll leave it in. But I still think we should reject tests for this mode and direct our efforts into a proper solution for platforms where fswatch is needed.

Reply:
Agreed about tests. However the passive watch mode is available outside of tests. I'm also working on a `dune rpc wait` command to replace the passive watch mode and it will require the sync mechanism.

Reply:
This doesn't seem to have been reverted BTW